texas-hearing-institute/listening_settings/ListeningButtonGroup.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import { Pressable, View, Text, StyleSheet } from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+export default function ListeningButtonGroup() {
+    const [selectedMode, setSelectedMode] = useState("");
+
+    useEffect(() => {
+        // get set button in storage
+        AsyncStorage.getItem("listening_settings.selectedMode").then(r => {
+            if (r != null) {
+                setSelectedMode(r);
+            }
+        })
+    }, [])
+
+    const buttonSelectHandler = (name: string) => {
+        if (selectedMode === name) {
+            setSelectedMode("");
+        } else {
+            setSelectedMode(name);
+        }
+    }
+
+    useEffect(() => {
```

Comment:
again, instead of this `useEffect` call we can just handle this inside of `buttonSelectHandler` either directly or by writing a separate function and calling it in there. take a look at the [Async Storage docs](https://react-native-async-storage.github.io/async-storage/docs/usage/#storing-string-value)